### ISSUE TITLE
CRIU: EOF when attempting to take a checkpoint

### ISSUE THREAD

**Issue by tajila** [eclipse-openj9/openj9#17230]
```
 [OUT] start running script
 [OUT] export GLIBC_TUNABLES=glibc.cpu.hwcaps=-XSAVEC,-XSAVE,-AVX2,-ERMS,-AVX,-AVX_Fast_Unaligned_Load
 [OUT] export LD_BIND_NOT=on
 [OUT] Pre-checkpoint
 [OUT] Performing CRIUSupport.checkpointJVM(), current thread name: main, Wed Apr 19 13:59:34 EDT 2023, System.currentTimeMillis(): 1681927174800, System.nanoTime(): 1681927174795989099
 [OUT] Exception in thread "main" org.eclipse.openj9.criu.SystemCheckpointException: Could not dump the JVM processes, err=-52
 [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVMImpl(Native Method)
 [OUT] 	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVM(CRIUSupport.java:649)
 [OUT] 	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:77)
 [OUT] 	at org.openj9.criu.OptionsFileTest.jitOptionsTest(OptionsFileTest.java:294)
 [OUT] 	at org.openj9.criu.OptionsFileTest.main(OptionsFileTest.java:66)
 [OUT] Removed testOutput file
 [OUT] finished script
 [ERR] Error (criu/util.c:457): Can't read link of fd -404: No such file or directory
 [ERR] Error (criu/protobuf.c:75): Unexpected EOF on (null)
```

https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_s390x_linux_Personal/283/tapResults/

**Comment by tajila** [comment]
Just making an issue here to track it, the most recent failure is on s390x

**Comment by ymanton** [comment]
The failures I've looked at are caused by a PPC-specific bug in CRIU 3.17.1. The stack that CRIU uses when running code in the process to be checkpointed is not set up correctly and doesn't comply with the ABI requirements. I fixed this bug upstream in https://github.com/checkpoint-restore/criu/commit/615763ec2d9777167f934f9e6a6830140a94b4ff but it is only available in the recently released 3.18 and later.

s390x does not have this problem so those failures are unrelated.
